### PR TITLE
Feature lazy pages

### LINKED ISSUE #135: Extract messages with card-actions to lancie-message-card
There are quite a few pages that still use the old copy-pasta paper-card with styling. They should be transformed into the new `lancie-message-card`. This requires changes in the element, as they now require `card-actions` to be added.

### LINKED ISSUE #283: Rework client architecture to be compatible with polymer-build
The current client architecture is incompatible with Polymer build. To be able to generate a PRPL-ready application, several changes are required:
- [ ] Rework `lancie-content` to allow lazy-loading of pages for faster first-paint
  - Adopt `iron-lazy-pages` to lazy-load pages
  - Check all imports to be correctly resolved
- [ ] Adopt a service-worker for faster cache results for static files (I should have a compatible one lying around somewhere)
- [x] Move files in `app/` to `src/` or `/`. All files should be served from `/` instead of `app/`, which means that files like `index.html` and `manifest.json` need to be in root. However, all other elements can remain in `src/` (which is a simple remain to be consistent with all other frontend projects)
- [x] Replace `page.js` with `app-route`

This is all based on https://github.com/PolymerElements/generator-polymer-init-custom-build and similar efforts I already did on other projects.

### PR DESCRIPTION
**Pull request by martijnjanssen**
Fixes #135 
Partially fixes #283 
Lazy pages fix a whole lot! Unfortunately not all is fixed yet, there are some difficulties implementing lazy-pages. The things that need to be looked at in this PR are: 
- [x] `paper-dialog` elements being positioned behind the optional backdrop
- [x] `<a>` elements do not have default styling for some reason. This may be due to the fact that it is not correctly applied at loading
- [x] I have resorted to using a workaround to fix the height of an image in a `paper-card` element. The header can contain an image, this image will have a minimum height of `600px` for some reason. This is now fixed by setting it to `100%`

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
-    "google-youtube": "GoogleWebComponents/google-youtube#~1.1.2",
+    "google-map": "GoogleWebComponents/google-map#~1.1.14",
     "moment": "momentjs#~2.10.6",
-    "polymer": "polymer/polymer#^1.4.0",
+    "polymer": "polymer/polymer#^1.7.0",
     "wisvch-footer": "wisvch/wisvch-footer#^1.0.1",
     "app-layout": "PolymerElements/app-layout#^0.10.4",
-    "app-route": "polymerelements/app-route#^0.9.2"
+    "app-route": "polymerelements/app-route#^0.9.2",
+    "iron-lazy-pages": "TimvdLippe/iron-lazy-pages"
```
Comment (TimvdLippe) [inline bower.json]: `#^1.0.0`

File: src/lancie-card.html [hunk 2]
```diff
@@ -24,21 +24,22 @@
...
         }
 
+        --paper-card-header-image: {
+          height: 100%;
+        }
       }
     </style>
 
     <paper-card heading="{{heading}}" image="https://areafiftylan.nl/images/logos/lancie/logo_without_date_with_background.png">
-      <div class="card-content">
```
Comment (TimvdLippe) [inline src/lancie-card.html]: Why was this div removed? This should be stamped inside the paper card content.

File: src/lancie-content.html [hunk 3]
```diff
@@ -43,92 +43,83 @@
...
         </lancie-login-check>
-      </div>
+      </template>
 
-      <div data-route="transfer-success">
-        <!-- This is done to prevent firing this request when it is not supposed to fire. -->
-        <lancie-success-ticket-transfer></lancie-success-ticket-transfer>
-      </div>
+      <template is="iron-lazy-page" data-route="transfer-success">
+            <lancie-success-ticket-transfer></lancie-success-ticket-transfer>
```
Comment (TimvdLippe) [inline src/lancie-content.html]: Spacing

File: src/lancie-my-area/lancie-accept-ticket-transfer.html [hunk 4]
```diff
@@ -85,7 +90,7 @@
       onAcceptTicketTransfer: function(event, r) {
         this.transferring = false;
         if (r.succeeded) {
-          window.location.href = '/#!/transfer-success';
+          window.location.href = '/transfer-success';
```
Comment (TimvdLippe) [inline src/lancie-my-area/lancie-accept-ticket-transfer.html]: Oof I did not see this one. We really should fix this the Polymer way. One solution is to fire an event and catch it in lancie-content to set the selected route.

File: src/lancie-my-area/my-area-discord.html [hunk 5]
```diff
@@ -46,7 +54,9 @@
           </div>
         </div>
       </div>
-      <paper-button raised on-tap="_openDiscord">Join!</paper-button>
+      <div class="card-actions">
```
Comment (TimvdLippe) [inline src/lancie-my-area/my-area-discord.html]: :+1:

File: bower.json (hunk 1 above)
Comment (mgajanssen) [inline bower.json]: This release does not seem to exist on their (github)[https://github.com/PolymerElements/app-route/releases} page, so will this work?

Comment (TimvdLippe) [inline bower.json]: The version is of my element, not app-route ;)

File: src/lancie-card.html (hunk 2 above)
Comment (mgajanssen) [inline src/lancie-card.html]: This is removed to enable us to also insert `card-actions` into the `paper-card`.

File: src/lancie-my-area/lancie-accept-ticket-transfer.html (hunk 4 above)
Comment (mgajanssen) [inline src/lancie-my-area/lancie-accept-ticket-transfer.html]: I think that is a better solution to solve this problem. This also makes changing the route in the javascript overall a better option.

File: src/lancie-content.html [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <link rel="import" href="../bower_components/iron-ajax/iron-ajax.html">
-<link rel="import" href="../bower_components/iron-pages/iron-pages.html">
 <link rel="import" href="../bower_components/app-route/app-location.html">
 <link rel="import" href="../bower_components/app-route/app-route.html">
+<link rel="import" href="../bower_components/iron-lazy-pages/iron-lazy-pages.html">
 <link rel="import" href="lancie-login/lancie-ajax.html">
 <link rel="import" href="lancie-login/lancie-login-check.html">
 <link rel="import" href="lancie-login/lancie-register-confirmation.html">
```
Comment (TimvdLippe) [inline src/lancie-content.html]: The imports of all elements inside routes should be lazyloaded. Thats the power of iron-lazy-pages :)

Comment (martijnjanssen) [inline src/lancie-content.html]: I do not quite understand what you mean, could you elaborate?

File: src/lancie-ticket-page/lancie-ticket-page.html [hunk 7]
```diff
@@ -95,7 +95,9 @@
     <lancie-ajax id="ajax" refurl="orders/{{order.id}}/checkout" on-response="goToPayment"></lancie-ajax>
     <lancie-ajax auto refurl="tickets/available" on-response="loadTickets"></lancie-ajax>
     <lancie-ajax auto refurl="users/current/orders/open" method="GET" on-response="getOpenOrder"></lancie-ajax>
+
     <template is="dom-if" if="{{canOrderTickets}}" restamp="true">
+    test
```
Comment (TimvdLippe) [inline src/lancie-ticket-page/lancie-ticket-page.html]: Leftover 😄

File: bower.json [hunk 8]
```diff
@@ -2,18 +2,18 @@
   "name": "lancie-frontend",
   "private": true,
   "dependencies": {
-    "iron-elements": "PolymerElements/iron-elements#^1.0.0",
-    "paper-elements": "PolymerElements/paper-elements#^1.0.0",
-    "platinum-elements": "PolymerElements/platinum-elements#^1.0.0",
+    "iron-elements": "PolymerElements/iron-elements#^1.0.10",
+    "paper-elements": "PolymerElements/paper-elements#^1.0.7",
+    "platinum-elements": "PolymerElements/platinum-elements#^2.0.0",
```
Comment (TimvdLippe) [inline bower.json]: Why did you bump this to a major version?

Comment (TimvdLippe) [inline bower.json]: Actually, are we even using them? :smile:

Comment (martijnjanssen) [inline bower.json]: Nope, I have no clue why they are in there.

**Pull request merged**